marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/savedQueries/SavedQueriesServiceTest.java

```diff
@@ -0,0 +1,192 @@
+                "          \"dataType\": \"string\",\n" +
+                "          \"stringValues\": [\n" +
+                "            \"Entity1\",\n" +
+                "            \"Collection1\"\n" +
+                "          ]\n" +
+                "        },\n" +
+                "        \"facet1\": {\n" +
+                "          \"dataType\": \"decimal\",\n" +
+                "          \"rangeValues\": {\n" +
+                "            \"lowerBound\": \"2.5\",\n" +
+                "            \"upperBound\": \"15\"\n" +
+                "          }\n" +
+                "        },\n" +
+                "        \"facet2\": {\n" +
+                "          \"dataType\": \"dateTime\",\n" +
+                "          \"rangeValues\": {\n" +
+                "            \"lowerBound\": \"2020-01-01T13:06:17\",\n" +
+                "            \"upperBound\": \"2020-01-22T13:06:17\"\n" +
+                "          }\n" +
+                "        }\n" +
+                "      }\n" +
+                "    },\n" +
+                "    \"propertiesToDisplay\": [\"facet1\", \"EntityTypeProperty1\"]\n" +
+                "  }\n" +
+                "}";
+        ObjectMapper mapper = new ObjectMapper();
+        queryDoc = mapper.readTree(jsonString);
+    }
+
+    @Test
+    void testSaveAndModifyQuery() {
+        JsonNode firstResponse = savedQueriesService.saveSavedQuery(queryDoc);
+
+        Assertions.assertNotNull(firstResponse);
+        Assertions.assertNotNull(firstResponse.get("savedQuery"));
+        Assertions.assertNotNull(firstResponse.get("savedQuery").get("id"));
+        Assertions.assertNotNull(firstResponse.get("savedQuery").get("systemMetadata"));
+        Assertions.assertEquals("some-query", firstResponse.get("savedQuery").get("name").asText());
+        Assertions.assertEquals(4, firstResponse.get("savedQuery").get("systemMetadata").size());
+        Assertions.assertEquals("flow-developer", firstResponse.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("flow-developer", firstResponse.get("savedQuery").get("systemMetadata").get("createdBy").asText());
+
+        ObjectNode savedQueryNode = (ObjectNode) firstResponse.get("savedQuery");
+        String id = savedQueryNode.get("id").asText();
+        savedQueryNode.put("name", "modified-name");
+        JsonNode modifiedResponse = savedQueriesService.saveSavedQuery(firstResponse);
+
+        Assertions.assertNotNull(modifiedResponse);
+        Assertions.assertEquals(id, modifiedResponse.get("savedQuery").get("id").asText());
+        Assertions.assertEquals("flow-developer", modifiedResponse.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("modified-name", modifiedResponse.get("savedQuery").get("name").asText());
+    }
+
+    @Test
+    void testCollectionsAndPermissionOnSavedAndModifiedQuery() {
+        JsonNode firstResponse = savedQueriesService.saveSavedQuery(queryDoc);
```

Comment:
I really recommend combining these assertions with the first test. The setup of the test is identical - you're saving a single saved query. You then want to assert on the contents of the persisted document, along with its URI and collections/permissions. There's no reason to do that in two separate tests, it just makes our test suite take a little longer to finish.